### DISCUSSION TITLE
24bit tracks shown as 2 bit in player

### DISCUSSION THREAD

**Discussion by marcelgsteiger** [rern/rAudio#1534] [Bug Reports]
Hi all
This might be a cosmetic issue. When I play 96kHz 24bit FLAC tracks, the player shows them as 2 bit.
I have two players showing the same issue, irrespective of the software version (currently using 20231216).
![screenshot](https://github.com/rern/rAudio/assets/11504328/2e6cbb8c-95f0-41b6-8c98-f0c8872241a3)
This is perhaps only a cosmetic issue - everything is working perfectly.

**Comment by cmh714** [comment]
Almost 100% sure its just a UI thingy....the regex is a little off :)

**Reply by klassikmann** [reply]
See https://github.com/rern/rAudio/discussions/1533#discussion-5968334 No. 2.

**Reply by ErikDorstel** [reply]
I found the cause. That is the fix:

/srv/http/bash/status.sh  ---  line:185

change:
			`samplerate=${val/:*}`
			`bitdepth=${val/*:}`

to:
			`data=( ${val//:/ } )`
			`samplerate=${data[0]}`
			`bitdepth=${data[1]}`

**Reply by klassikmann** [reply]
@ErikDorstel 
Your patch seems to work. Thank you.
From my point of view even more confusing is my comment No 1 from https://github.com/rern/rAudio/discussions/1533#discussion-5968334

**Reply by ErikDorstel** [reply]
I cannot (re)produce the issue from comment 1.

(My suggested change to correct the bitrate was a mistake. I deleted it.)

**Reply by ErikDorstel** [reply]
fix as pull request

https://github.com/rern/rAudio/pull/1540

**Comment by ErikDorstel** [comment]
I have a lot of HD music larger than 16 bit. That's why the incorrect display affects me greatly.

I'm trying to locate the problem. I exclude the client UI because the files in /srv/http/data/shm/sampling also contain invalid values for the bitdepth.

Right now I'm trying to understand status.sh. Two methods are used to examine the media files. Firstly, using a Telnet connection to the MPD and the "playlistinfo" command. On the other hand, with the shell command "ffprobe". But in both cases I can't find the place in the code where the bitdepth is read.

Does anyone have more detailed knowledge of the function?
